Synview-frontend/src/components/MentorInviteForm.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import { useInviteMentorMutation } from "../services/apiSlice.ts";
+import { useAppSelector } from "../hooks.ts";
+import { ProjectRoles } from "../../../common/schemas.ts";
+export default function MentorInviteForm() {
+  const [invitedMentorId, setInvitedMentorId] = useState("");
+
+  const project_id = useAppSelector(
+    (state) => state.inviteMentorModal.project_id
+  );
+  const user_id = useAppSelector((state) => state.inviteMentorModal.user_id);
+  const [inviteMentor] = useInviteMentorMutation();
+  const handleInviteForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      if (project_id && user_id) {
+        await inviteMentor({
```

Comment:
May want a loading state that disables the button, otherwise the user can click multiple times. Using an abstracted query wrapper that handles loading and errors can help with this.

Reply:
Looks like this comes from `useInviteMentorMutation`; the second argument has a `isLoading` prop which serves this need. It also has `error` which could allow you to render a generic error banner or similar.